test/benchmarks/squad_to_dpr.py

```diff
@@ -0,0 +1,320 @@
+                positive_ctxs = [{
+                    "title": f"{article_title}_{i}",
+                    "text": c
+                } for i, c in enumerate([context for _ in question["answers"]])]
+
+                if not hard_negative_ctxs or not positive_ctxs:
+                    logging.error(
+                        f"No retrieved candidates for article {article_title}, with question {question['question']}")
+                    n_non_added_questions += 1
+                    continue
+                dict_DPR = {
+                    "question": question["question"],
+                    "answers": answers,
+                    "positive_ctxs": positive_ctxs,
+                    "negative_ctxs": [],
+                    "hard_negative_ctxs": hard_negative_ctxs
+                }
+                n_questions += 1
+                yield dict_DPR
+
+    print(f"Number of not added questions : {n_non_added_questions} / {n_questions}")
+
+
+def split_and_save_dataset(iter_dpr: Iterator, dpr_output_path: Path,
+                           total_nb_questions: int):
+    nb_train_examples = int(total_nb_questions * 0.8)
+    nb_dev_examples = int(total_nb_questions * 0.1)
+
+    train_iter = islice(iter_dpr, nb_train_examples)
+    dev_iter = islice(iter_dpr, nb_dev_examples)
+
+    dataset_splits = {
+        dpr_output_path / Path("DPR_train.json"): train_iter,
+        dpr_output_path / Path("DPR_dev.json"): dev_iter,
+        dpr_output_path / Path("DPR_test.json"): iter_dpr,
+    }
+
+    for path, set_iter in dataset_splits.items():
+        with open(path, "w") as json_ds:
+            json.dump(IteratorAsList(set_iter), json_ds, indent=4)
+
+
+def get_hard_negative_context(retriever: BaseRetriever, question: str, answer: str,
+                              n_ctxs: int = 30):
+    list_hard_neg_ctxs = []
+    retrieved_docs = retriever.retrieve(query=question, top_k=n_ctxs, index="document")
+    for retrieved_doc in retrieved_docs:
+        retrieved_doc_id = retrieved_doc.meta["name"]
+        retrieved_doc_text = retrieved_doc.text
+        if answer.lower() in retrieved_doc_text.lower():
+            continue
+        list_hard_neg_ctxs.append({"title": retrieved_doc_id, "text": retrieved_doc_text})
+
+    return list_hard_neg_ctxs
+
+
+def load_squad_file(squad_file_path: Path):
+    squad_data = json.load(open(squad_file_path.as_posix()))
+    return squad_data
+
+
+def main(squad_file_path: Path, dpr_output_path: Path,
+         document_store_type_config: Tuple[str, Dict] = ("ElasticsearchDocumentStore", {}),
+         retriever_type_config: Tuple[str, Dict] = ("ElasticsearchRetriever", {}),
+         num_hard_negative_ctxs: int = 30):
+    tqdm.write(f"Using SQuAD-like file {squad_file_path}")
+
+    # 1. Load squad file data
+    squad_impossible_path, squad_data = add_is_impossible(load_squad_file(squad_file_path=squad_file_path),
```

Comment:
We are generally using datasets in Squad v2 format where the `is_impossible` field already exists. Let's have a check here to see if the `is_impossible` field exists already. If it doesn't (e.g. Squad1) we'll call `add_is_impossible` to add it. Otherwise, let's bypass it.

Reply:
Also, we should skip cases where `is_impossible=True`

Reply:
same as before. Will add this check.